### PR TITLE
IW-130 | User-Intended plays recognition module

### PR DESCRIPTION
**Pull request by bkoval**
@Wikia/iwing

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -91,7 +91,6 @@ function wikiaJWPlayerEvents(playerInstance, willAutoplay, logger) {
 		});
 
 		relatedPlugin.on('play', function (data) {
-			logger.info('related plugin play');
```
Comment (bkoval) [inline src/events.js]: Looks like a dev leftover from the past, cleanups.

File: src/index.js [hunk 2]
```diff
@@ -154,6 +155,14 @@ window.wikiaJWPlayer = function (elementId, options, callback) {
 
 		wikiaJWPlayerAllowControllOnTouchDevices(playerInstance);
 
+		wikiaJWPlayerUserIntendedPlayControl(
+			//user coming from footer must've clicked the video thumbnail
+			window.location.search.indexOf('wikia-footer-wiki-rec') > -1,
```
Comment (rybmat) [inline src/index.js]: not sure if we should do it now, but how about extracting list of such queryparameters to some array? I assume there will be more places we would count as UIP

File: src/user-intended-play-control.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		}
+	}
+
+	function onPause(data) {
+		if (data.pauseReason === 'interaction') {
+			wasPausedByUserInteraction = true;
+		}
+	}
+
+	function onPlay(data) {
```
Comment (latata) [inline src/user-intended-play-control.js]: Unused `data`

File: src/user-intended-play-control.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	function onFullScreen() {
+		setUserIntendedPlay(true, reasonsForUserIntendedPlay.fullScreen);
+	}
+
+	function onUnmute() {
+		setUserIntendedPlay(true, reasonsForUserIntendedPlay.unmuting);
+	}
+
+	function onMute() {
+		if (isUserIntendedPlay && userIntendedPlayReason === reasonsForUserIntendedPlay.unmuting) {
```
Comment (latata) [inline src/user-intended-play-control.js]: As we talked offline please get rid of `reasonsForUserIntendedPlay` and add flag which determines if the reason of user intended play was unmuting. Or even we can get rid of this `if` completely. Ask product.

File: src/user-intended-play-control.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		} else {
+			userIntendedReason = null;
+		}
+
+		if (typeof tracker.setCustomDimension !== 'function') {
+			return;
+		}
+
+		if (immediate) {
+			tracker.setCustomDimension(39, isUserIntended ? 'user-intended' : 'not-user-intended');
```
Comment (latata) [inline src/user-intended-play-control.js]: lets extract CD number and values to consts

File: src/index.js (hunk 2 above)
Comment (bkoval) [reply src/index.js]: That makes sense but we I will actually move this logic to app repository and implement it there :)

**Pull request merged**